File: coyote/apis/auth_token_namespace.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    AuthTokenCreateNamespaceIn,
    AuthTokenCreateNamespaceOut,
    AuthTokenGetNamespaceIn,
    AuthTokenGetNamespaceOut,
)


class AuthTokenNamespaceAsync(ApiBase):
    async def create(
        self,
        auth_token_create_namespace_in: AuthTokenCreateNamespaceIn,
    ) -> AuthTokenCreateNamespaceOut:
        """Create Auth Token namespace"""
        body = auth_token_create_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.namespace.create",
            body=body,
            response_type=AuthTokenCreateNamespaceOut,
        )

    async def get(
        self,
        auth_token_get_namespace_in: AuthTokenGetNamespaceIn,
    ) -> AuthTokenGetNamespaceOut:
        """Get Auth Token namespace"""
        body = auth_token_get_namespace_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.namespace.get",
            body=body,
            response_type=AuthTokenGetNamespaceOut,
        )


class AuthTokenNamespace(ApiBase):
    def create(
        self,
        auth_token_create_namespace_in: AuthTokenCreateNamespaceIn,
    ) -> AuthTokenCreateNamespaceOut:
        """Create Auth Token namespace"""
        body = auth_token_create_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.namespace.create",
            body=body,
            response_type=AuthTokenCreateNamespaceOut,
        )

    def get(
        self,
        auth_token_get_namespace_in: AuthTokenGetNamespaceIn,
    ) -> AuthTokenGetNamespaceOut:
        """Get Auth Token namespace"""
        body = auth_token_get_namespace_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.namespace.get",
            body=body,
            response_type=AuthTokenGetNamespaceOut,
        )

File: src/api/auth_token_namespace.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct AuthTokenNamespace<'a> {
    cfg: &'a Configuration,
}

impl<'a> AuthTokenNamespace<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Create Auth Token namespace
    pub async fn create(
        &self,
        auth_token_create_namespace_in: AuthTokenCreateNamespaceIn,
    ) -> Result<AuthTokenCreateNamespaceOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.namespace.create")
            .with_body(auth_token_create_namespace_in)
            .execute(self.cfg)
            .await
    }

    /// Get Auth Token namespace
    pub async fn get(
        &self,
        auth_token_get_namespace_in: AuthTokenGetNamespaceIn,
    ) -> Result<AuthTokenGetNamespaceOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.namespace.get")
            .with_body(auth_token_get_namespace_in)
            .execute(self.cfg)
            .await
    }
}
